resources/util.ts

```diff
@@ -176,6 +181,141 @@ const watchCombatant: WatchCombatantFunc = (params, func) => {
   return defaultWatchCombatant(params, func);
 };
 
+export type DirectionOutput =
+  | 'dirN'
+  | 'dirNNE'
+  | 'dirNE'
+  | 'dirENE'
+  | 'dirE'
+  | 'dirESE'
+  | 'dirSE'
+  | 'dirSSE'
+  | 'dirS'
+  | 'dirSSW'
+  | 'dirSW'
+  | 'dirWSW'
+  | 'dirW'
+  | 'dirWNW'
+  | 'dirNW'
+  | 'dirNNW'
+  | 'unknown';
+
+export type DirectionLabel =
+  | 'N'
+  | 'NNE'
+  | 'NE'
+  | 'ENE'
+  | 'E'
+  | 'ESE'
+  | 'SE'
+  | 'SSE'
+  | 'S'
+  | 'SSW'
+  | 'SW'
+  | 'WSW'
+  | 'W'
+  | 'WNW'
+  | 'NW'
+  | 'NNW'
+  | '???';
+
+export const output8Dir: DirectionOutput[] = [
+  'dirN',
+  'dirNE',
+  'dirE',
+  'dirSE',
+  'dirS',
+  'dirSW',
+  'dirW',
+  'dirNW',
+];
+export const outputCardinalDir: DirectionOutput[] = ['dirN', 'dirE', 'dirS', 'dirW'];
```

Comment:
> Sorry to be picky here, but one thing I think would be useful would be to have also a set of output strings, e.g. `outputStringsDir8` or `outputStringsDIr16` that can be used like `...Util.dir.outputStringsDir8` or whatever rather than listing them all manually.

I was actually thinking the same.  Figured I would add it once I started getting into the fight scripts, but no reason not to do it now.

> But, in that case, I think it would be better to have `DirectionOutput8` vs `DirectionOutput16` types because `addedCombatantPosTo8DirOutput` can't ever output `dirNNE`.

Assuming I add `outputStringsDir8` and `outputStringsDIr16`, is there a reason why they can't both use the type `DirectionOutput` as opposed to defining two separate types where `DirectionOutput8` is essentially just a subset of `DirectionOutput16`?  Not disagreeing, just want to understand your thinking better as I start in on some of the fight scripts.

Reply:
Mostly it's been on my mind to try to see if I can make `Output` be templated based on outputstring keys at some point.  (It's possible @valarnin already tried this or there's some complexity limit here.) So, mostly just thinking about that future; also, if a function can only return 8+unknown directions then it seems vaguely better to have a narrower type that represents that limitation.